### PR TITLE
[youtube] fixed --flat-playlist always showing uploader as none

### PR DESCRIPTION
**Pull request by PrinceOfPuppers**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

running:
```
youtube-dl --flat-playlist -j YOUTUBE_PLAYLIST_URL
```
prior to the bugfix would always have "uploader": none in each playlist item. post fix the proper uploader will appear in said field

### CODE REVIEW

**Comment by adrianheine** [comment]
Hey, thanks for your PR! Please prefix your commit message (and PR title) with `[youtube]` to indicate that it's about the youtube IE.

**Comment by PrinceOfPuppers** [comment]
> Hey, thanks for your PR! Please prefix your commit message (and PR title) with `[youtube]` to indicate that it's about the youtube IE.

woops sorry, updated.

**Comment by PrinceOfPuppers** [comment]
got rid of the extra 2 commits i accidentally created when attempting to rename the commit

File: youtube_dl/extractor/youtube.py [hunk 1]
```diff
@@ -308,7 +308,7 @@ def _extract_video(self, renderer):
             r'^([\d,]+)', re.sub(r'\s', '', view_count_text),
             'view count', default=None))
         uploader = try_get(
-            renderer, lambda x: x['ownerText']['runs'][0]['text'], compat_str)
```
Comment (dstftw) [inline youtube_dl/extractor/youtube.py:311]: Do not remove the old pattern.

**Review by dstftw — CHANGES_REQUESTED** [review]

**Comment by dstftw** [comment]
Add test.

File: test/test_youtube_lists.py [hunk 2]
```diff
@@ -66,6 +67,16 @@ def test_youtube_flat_playlist_titles(self):
...
 
+    def test_youtube_flat_playlist_uploaders(self):
+        dl = FakeYDL()
+        dl.params['extract_flat'] = True
+        ie = YoutubeTabIE(dl)
+        result = ie.extract('https://www.youtube.com/playlist?list=PL2zzFmCbIz4fshCLrvopue-wDeKQ5ERKO')
+        self.assertIsPlaylist(result)
+        for entry in result['entries']:
+            if entry['title'] not in ('[Deleted video]', '[Private video]'):
+                self.assertTrue(entry.get('uploader'))
```
Comment (dstftw) [inline test/test_youtube_lists.py:78]: Test should be added to `_TESTS` of extractor class. Use one of the youtube-dl's test playlists for testing, e.g. https://www.youtube.com/playlist?list=PL4lCao7KL_QFVb7Iudeipvc2BCavECqzc.

**Review by dstftw — CHANGES_REQUESTED** [review]

**Pull request closed without merging**